recipe/build.sh

```diff
@@ -26,6 +26,10 @@ export USE_NUMA=0
 export USE_ITT=0
 
 #################### ADJUST COMPILER AND LINKER FLAGS #####################
+# Pytorch's build system doesn't like us setting the c++ standard and will
+# issue a warning:
+# https://github.com/pytorch/pytorch/blob/3beb7006dd5a415dfa236081ad5d55ae38346324/CMakeLists.txt#L41
+export CXXFLAGS="$(echo $CXXFLAGS | sed 's/-std=c++[0-9][0-9]//g')"
```

Comment:
We need to ensure that this is C++17 for reasons of abseil's ABI, see https://github.com/conda-forge/abseil-cpp-feedstock/issues/45. Luckily, pytorch 2.5 is on C++17 [already](https://github.com/pytorch/pytorch/blob/v2.5.1/CMakeLists.txt#L42) (and I agree with pytorch that this shouldn't be done through compiler flags but through `CMAKE_CXX_STANDARD`). Furthermore, conda-forge doesn't set the C++ standard in the compiler activation, so there's little chance that we see this in `CXXFLAGS`, but in any case, this change should be benign.

Reply:
ok, fair. Anaconda does set the standard via `CXXFLAGS` but also sets it to 17. I think I had some issues with this in the past so it's probably safest to keep it in, but if you prefer us to keep it on our side I'm fine with that.

Reply:
left as-is